### PR TITLE
8320147: Remove DumpSharedSpaces

### PR DESCRIPTION
**Pull request by iklam**
One more PR for cleanup with cdsConfig.hpp:

Replace the global variable `DumpSharedSpaces` with `CDSConfig::is_dumping_static_archive()`.

Note: some mis-uses of `DumpSharedSpaces` need to be replaced with `CDSConfig::is_dumping_heap()` or `CDSConfig::is_dumping_full_module_graph()`

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8320147](https://bugs.openjdk.org/browse/JDK-8320147): Remove DumpSharedSpaces (**Sub-task** - P4)

### Reviewers
 * [Calvin Cheung](https://openjdk.org/census#ccheung) (@calvinccheung - **Reviewer**) ⚠️ Review applies to [613b5de2](https://git.openjdk.org/jdk/pull/16700/files/613b5de2ebfa86f4121e29fa7cf16e4e4edc54d9)
 * [Matias Saavedra Silva](https://openjdk.org/census#matsaave) (@matias9927 - Committer) ⚠️ Review applies to [613b5de2](https://git.openjdk.org/jdk/pull/16700/files/613b5de2ebfa86f4121e29fa7cf16e4e4edc54d9)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/16700/head:pull/16700` \
`$ git checkout pull/16700`

Update a local copy of the PR: \
`$ git checkout pull/16700` \
`$ git pull https://git.openjdk.org/jdk.git pull/16700/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 16700`

View PR using the GUI difftool: \
`$ git pr show -t 16700`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/16700.diff">https://git.openjdk.org/jdk/pull/16700.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/16700#issuecomment-1815432217)

### CODE REVIEW

File: src/hotspot/share/oops/instanceClassLoaderKlass.cpp [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 1997, 2023, Oracle and/or its affiliates. All rights reserved.
```
Comment (calvinccheung) [inline src/hotspot/share/oops/instanceClassLoaderKlass.cpp:2]: Since this is a new file, no need to include 1997.

**Review by calvinccheung — APPROVED** [review]
Few files require copyright header update.

instanceClassLoaderKlass.hpp
instanceMirrorKlass.hpp
instanceRefKlass.hpp
instanceStackChunkKlass.hpp

**Review by matias9927 — APPROVED** [review]
LGTM

**Comment by iklam** [comment]
Thanks @calvinccheung and @matias9927 for the review.
/integrate

**Pull request closed without merging**